si5351/si5351.go

```diff
@@ -0,0 +1,443 @@
+
+	data = d.buf[:8]
+
+	// Power down all output drivers
+	for i := range data {
+		data[i] = 0x80
+	}
+	d.bus.WriteRegister(d.Address, CLK0_CONTROL, data)
+
+	// Disable spread spectrum output.
+	d.DisableSpreadSpectrum()
+
+	d.initialised = true
+
+}
+
+// Connected returns whether a device at SI5351 address has been found.
+func (d *Device) Connected() bool {
+	err := d.bus.Tx(uint16(d.Address), []byte{}, []byte{0})
+	return err == nil
+}
+
+func (d *Device) EnableSpreadSpectrum() (err error) {
+	data := d.buf[:1]
+	err = d.bus.ReadRegister(d.Address, SPREAD_SPECTRUM_PARAMETERS, data)
+	if err != nil {
+		return
+	}
+	data[0] |= 0x80
+	err = d.bus.WriteRegister(d.Address, SPREAD_SPECTRUM_PARAMETERS, data)
+	return
+}
+
+func (d *Device) DisableSpreadSpectrum() (err error) {
+	data := d.buf[:1]
+	err = d.bus.ReadRegister(d.Address, SPREAD_SPECTRUM_PARAMETERS, data)
+	if err != nil {
+		return
+	}
+	data[0] &^= 0x80
+	err = d.bus.WriteRegister(d.Address, SPREAD_SPECTRUM_PARAMETERS, data)
+	return
+}
+
+func (d *Device) EnableOutputs() (err error) {
+	if !d.initialised {
+		return errNotInitialised
+	}
+	data := d.buf[:1]
+	data[0] = 0x00
+	err = d.bus.WriteRegister(d.Address, OUTPUT_ENABLE_CONTROL, data)
+	return
+}
+
+func (d *Device) DisableOutputs() (err error) {
+	if !d.initialised {
+		return errNotInitialised
+	}
+	data := d.buf[:1]
+	data[0] = 0xFF
+	err = d.bus.WriteRegister(d.Address, OUTPUT_ENABLE_CONTROL, data)
+	return
+}
+
+// ConfigurePLL sets the multiplier for the specified PLL
+// pll   The PLL to configure, which must be one of the following:
+// - PLL_A
+// - PLL_B
+//
+// mult  The PLL integer multiplier (must be between 15 and 90)
+//
+// num   The 20-bit numerator for fractional output (0..1,048,575).
+// Set this to '0' for integer output.
+//
+// denom The 20-bit denominator for fractional output (1..1,048,575).
+// Set this to '1' or higher to avoid divider by zero errors.
+//
+// PLL Configuration
+// fVCO is the PLL output, and must be between 600..900MHz, where:
+//
+// fVCO = fXTAL * (a+(b/c))
+//
+// fXTAL = the crystal input frequency
+// a     = an integer between 15 and 90
+// b     = the fractional numerator (0..1,048,575)
+// c     = the fractional denominator (1..1,048,575)
+//
+// NOTE: Try to use integers whenever possible to avoid clock jitter
+// (only use the a part, setting b to '0' and c to '1').
+//
+// See: http://www.silabs.com/Support%20Documents/TechnicalDocs/AN619.pdf
+func (d *Device) ConfigurePLL(pll uint8, mult uint8, num uint32, denom uint32) (err error) {
+
+	// Basic validation
+	if !d.initialised {
+		return errNotInitialised
+	}
+	// mult = 15..90
+	if !((mult > 14) && (mult < 91)) {
+		return errInvalidParameter
+	}
+	// Avoid divide by zero
+	if !(denom > 0) {
+		return errInvalidParameter
+	}
+	// 20-bit limit
+	if !(num <= 0xFFFFF) {
+		return errInvalidParameter
+	}
+	// 20-bit limit
+	if !(denom <= 0xFFFFF) {
+		return errInvalidParameter
```

Comment:
Here is a case where returning an unexported variable can case trouble: a caller might wish to do something like
if errors.Is(err, si5351.ErrInvalidParametr)
also, it's common practice nowadays  to wrap errors with more information. You have three ways at least to return an invalid parameter here and that can be very frustrating to debug. 
At the very least:
return fmt.Errorf("%#x is < 0xfffff: %w", denom, ErrInvalidParameter)

you may have to use %v of do something else if tinygo is not quite caught up, but this is a more useful error return pattern to use.